core/ui_events.js

```diff
@@ -113,5 +114,63 @@ Blockly.Events.Click.prototype.fromJson = function(json) {
   this.blockId = json['blockId'];
 };
 
+/**
+ * Class for a Selected event.
+ * @param {?string=} opt_oldElementId The id of the previously selected
```

Comment:
Do null and undefined mean different things for these two parameters (e.g. null if no last selected element, undefined if...?)? It's slightly confusing that the last one is only optional, but the first two allow both undefined and null. The Click event in this file doesn't allow the optional string parameters to be null, so I'd favor treating them the same unless null and undefined have different meanings.

Reply:
I've updated the param description. It is different yes. Null means that there was no last selected element and undefined means that it's a blank event.

Reply:
Cool, thanks for the explanation!